test/langtools/jdk/internal/shellsupport/doc/SinceCheckerHelper.java

```diff
@@ -0,0 +1,317 @@
+                        mappedVersion.introducedPreview() :
+                        mappedVersion.introducedStable();
+                checkEquals(sinceVersion, realMappedVersion, uniqueId);
+            } catch (Exception e) {
+                System.err.println("Error for " + uniqueId + " " + e.getMessage());
+            }
+            return sinceVersion;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean isPreview(Element el, String uniqueId, String currentVersion) {
+        while (el != null) {
+            Symbol s = (Symbol) el;
+            if ((s.flags() & Flags.PREVIEW_API) != 0) {
+                return true;
+            }
+            el = el.getEnclosingElement();
+        }
+        boolean legacyPreview = LEGACY_PREVIEW_METHODS.contains(uniqueId)
+                &&
+                (JDK13.equals(currentVersion) || JDK14.equals(currentVersion));
+        return legacyPreview;
+    }
+
+    public Version checkElement(TypeElement clazz, Element element, Types types,
+                                JavadocHelper javadocHelper, String currentVersion, Version enclosingVersion) {
+        String uniqueId = getElementName(clazz, element, types);
+        return checkElement(javadocHelper, uniqueId, currentVersion, enclosingVersion, element);
+    }
+
+    private Version extractSinceVersion(String documentation) {
+        Pattern pattern = Pattern.compile("@since\\s+(\\d+(?:\\.\\d+)?)");
+        Matcher matcher = pattern.matcher(documentation);
+        if (matcher.find()) {
+            String versionString = matcher.group(1);
+
+            if (versionString.equals("1.0")) {
+                //XXX
+                versionString = "1";
+            } else if (versionString.startsWith("1.")) {
+                versionString = versionString.substring(2);
+            }
+
+            try {
+                return Version.parse(versionString);
+            } catch (NumberFormatException ex) {
+                System.err.println("@since value that cannot be parsed: " + versionString);
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+
+    private void checkEquals(Version sinceVersion, String mappedVersion, String elementSimpleName) {
+        try {
+            //      System.err.println("For  Element: " + simpleName);
+            //      System.err.println("sinceVersion: " + sinceVersion + "\t mappedVersion: " +mappedVersion);
+            if (sinceVersion == null) {
+                return;
+            }
+            if (mappedVersion == null) {
+                System.out.println("check for why mapped version is null for" + elementSimpleName);
+                return;
+            }
+            if (Version.parse("9").compareTo(sinceVersion) > 0) {
+                sinceVersion = Version.parse("9"); //TODO: handle baseline version better
```

Comment:
Yes - please handle baseline better. Probably should say something like "existed before 10", or "introduced in JDK 9 or older", or something like that.